### PR TITLE
Add React Stack Component

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Adds the React Stack Component.

### Associated Issue 
Fixes https://github.com/sparkdesignsystem/spark-design-system/issues/1117.

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Documentation
 - [x] Update Spark Docs React

### Code
 - [x] Build Component in Spark React
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: src/patterns/fundamentals/grid/react/info/stack-object.hbs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      <th>Type</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">SprkStack</td>
+      <td>children</td>
+      <td>String</td>
```
Comment (corcorankt1) [inline src/patterns/fundamentals/grid/react/info/stack-object.hbs]: Should the type be "SprkStackItem" for children? SprkTabs children prop type is listed as SprkTabsPanel

File: src/react/projects/spark-core-react/src/SprkStack/SprkStack.test.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+});
+
+it('should set the tiny spacing class in addition to the base class when itemSpacing is "tiny"', () => {
+  const wrapper = shallow(
+    <SprkStack itemSpacing="tiny" />,
+  );
+  expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--tiny').length).toBe(1);
+});
+
+it('should set the tiny spacing class in addition to the base class when itemSpacing is "small"', () => {
```
Comment (corcorankt1) [inline src/react/projects/spark-core-react/src/SprkStack/SprkStack.test.js]: update to "small spacing"

File: src/react/projects/spark-core-react/src/SprkStack/SprkStack.test.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  const wrapper = shallow(
+    <SprkStack itemSpacing="small" />,
+  );
+  expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--small').length).toBe(1);
+});
+
+it('should set the tiny spacing class in addition to the base class when itemSpacing is "medium"', () => {
```
Comment (corcorankt1) [inline src/react/projects/spark-core-react/src/SprkStack/SprkStack.test.js]: medium spacing

File: src/react/projects/spark-core-react/src/SprkStack/SprkStack.test.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  const wrapper = shallow(
+    <SprkStack itemSpacing="medium" />,
+  );
+  expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--medium').length).toBe(1);
+});
+
+it('should set the tiny spacing class in addition to the base class when itemSpacing is "large"', () => {
```
Comment (corcorankt1) [inline src/react/projects/spark-core-react/src/SprkStack/SprkStack.test.js]: large spacing, etc etc

File: src/react/projects/spark-core-react/src/SprkStack/SprkStack.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      {children}
+    </div>
+  );
+};
+
+SprkStack.defaultProps = {
+  children: '',
+  splitAt: '',
+  itemSpacing: '',
+  idString: '',
```
Comment (bobwise) [inline src/react/projects/spark-core-react/src/SprkStack/SprkStack.js]: with a default value of `''`, the element will always render `data-id`. If you set the default value to `null`, the `data-id` attribute won't be added to the element unless it has a value.

Comment (afebbraro) [reply src/react/projects/spark-core-react/src/SprkStack/SprkStack.js]: We are going to chat about defaultProps in our next programming hour.

**Pull request merged**